File: employee-service/src/main/java/github/Jimoou/employeeservice/Entity/Employee.java
package github.Jimoou.employeeservice.Entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "employees")
public class Employee {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  private String firstName;
  private String lastName;

  @Column(nullable = false, unique = true)
  private String email;
}

File: react-frontend/src/model/Employee.ts
class Employee {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  departmentCode: string;
  organizationCode: string;

  constructor(
    id: number,
    firstName: string,
    lastName: string,
    email: string,
    departmentCode: string,
    organizationCode: string
  ) {
    this.id = id;
    this.firstName = firstName;
    this.lastName = lastName;
    this.email = email;
    this.departmentCode = departmentCode;
    this.organizationCode = organizationCode;
  }
}
export default Employee;
